src/controllers/post.js

```diff
@@ -328,31 +399,58 @@ export const deleteLike = async (req, res) => {
 export const setIsPrivate = async (req, res) => {
   const postId = Number(req.params.id)
 
-  try {
-    const foundPost = await dbClient.post.findUnique({
-      where: { id: postId }
-    })
-    if (!foundPost) {
-      return sendMessageResponse(
-        res,
-        404,
-        'The post with the provided id does not exist'
-      )
-    }
+  const foundPost = await dbClient.post.findUnique({
+    where: { id: postId }
+  })
+  if (!foundPost) {
+    myEmitter.emit(
+      'error',
+      req.user,
+      `update-post-${postId}-privacy`,
+      404,
+      'The post with the provided id does not exist'
+    )
```

Comment:
Classes would come in very handy for these events. Compare this line:

```js
emitter.emit('error', new PostNotFoundEvent(req.user, postId))
```

All of that specific errors details can be contained in that single class, you just pass it what you need at the time (which is only two parameters now)

```js
class PostNotFoundEvent {
    constructor(user, postId) {
        this.user = user
        this.postId = postId
        this.code = 404
        this.message = 'The post with the provided id does not exist'
        this.log = `update-post-${postId}-privacy`
    }
}
```

Reply:
Will do this to improve the code! Thanks for the advice!